docs/standard/garbage-collection/large-object-heap.md

```diff
@@ -0,0 +1,313 @@
+   };
+
+   Node[] binary_tr = new Node [num_nodes];
+   ```
+
+   If `num_nodes` is large, the garbage collector needs to go through at least two references per element. An alternative approach is to store the index of the right and the left nodes:
+
+   ```csharp
+   class Node
+   {
+      Data d;
+      uint left_index;
+      uint right_index;
+   } ;
+   ```
+
+   Instead of referring the left node’s data as `left.d`, you refer to it as `binary_tr[left_index].d`. And the garbage collector doesn’t need to look at any references for the left and right node.
+
+Out of the three factors, the first two are usually more significant than the third. Because of this, we recommend that you allocate a pool of large objects that you reuse instead of allocating temporary ones. 
+
+## Collecting performance data for the LOH
+
+Before you collect performance data for a specific area, you should already have done the following:
+
+1. Found evidence that you should be looking at this area.
+
+1. Exhausted other areas that you know of without finding anything that could explain the performance problem you saw.
+
+See the blog [Understand the problem before you try to find a solution](https://blogs.msdn.microsoft.com/maoni/2006/09/01/understand-the-problem-before-you-try-to-find-a-solution/) for more information on the fundamentals of memory and the CPU.
+
+You can use the following tools to collect data on LOH performance:
+
+- [.NET CLR memory performance counters](#net-clr-memory-performance-counters)
+
+- [ETW events](#etw-events)
+
+- [A debugger](#a-debugger)
+
+### .NET CLR Memory Performance counters.
+
+These performance counters are usually a good first step in investigating performance issues. You configure Performance Monitor by adding the counters that you want, as Figure 4 shows. The ones that are relevant for the LOH are:
+
+- **\# Gen 2 Collections**
+
+   Displays the number of times generation 2 GCs have occurred since the process started. The counter is incremented at the end of a generation 2 collection (also called a full garbage collection). This counter displays the last observed value.
+
+- **Large Object Heap size**
+
+   Displays the current size, in bytes, including free space, of the LOH. This counter is updated at the end of a garbage collection, not at each allocation.
+
+A common way to look at performance counters is with Performance Monitor (perfmon.exe). Use “Add Counters” to add the interesting counter for processes that you care about. You can save the performance counter data to a log file, as Figure 4 shows.
+
+![Figure 4: Adding performance counters.](media/loh/perfcounter.png)    
+Figure 4: The LOH after a generation 2 GC
+
+Performance counters can also be queried programmatically. Many people collect them this way as part of their routine testing process. When they spot counters with values that are out of the ordinary, they use other means to get more detailed data to help with the investigation.
+
+> [!NOTE]
```

Comment:
I'd consider switching these two sections, so we start with the recommended approach.